lib/Partitioner/Partitioner.cpp

```diff
@@ -423,19 +423,21 @@ void Partitioner::partitionsAdjust(NodeToFunctionMap &partitions,
     gain = false;
     for (FunctionToNodesMapTy::iterator it = nodesSet.begin();
          it != nodesSet.end(); ++it) {
-      NodesSetTy nSet = (*it).second;
-      std::vector<Node *> outUsers = getOutUsersWithOnePredecessor(nSet);
+      NodesSetTy *curSet = &((*it).second);
```

Comment:
Should it rather be `NodesSetTy &curSet`? Or do you plan to change the value of curSet pointer?

Reply:
Yes, the value maybe changed by inserting new node into this set.

Reply:
I'm not asking about the content of `NodesSetTy`. I'm saying if you assign to the NodesSetTy pointer only once, like in this line, you can replace it with a reference. But if you need to assign to the pointer multiple times, you really need a pointer.

Reply:
You are right!  No need to use the pointer now. Thanks!